src/schema.cpp

```diff
@@ -52,25 +52,25 @@ Schema::Schema(base types) : base(std::move(types))
     std::sort(begin(), end(), compare_by_name);
 }
 
-Schema::iterator Schema::find(std::string const& name)
+Schema::iterator Schema::find(StringData name)
 {
-    ObjectSchema cmp;
-    cmp.name = name;
-    return find(cmp);
+    auto it = std::lower_bound(begin(), end(), name, [](ObjectSchema const& lft, StringData rgt) {
+        return lft.name < rgt;
+    });
+    if (it != end() && it->name != name) {
+        it = end();
+    }
+    return it;
 }
 
-Schema::const_iterator Schema::find(std::string const& name) const
+Schema::const_iterator Schema::find(StringData name) const
 {
     return const_cast<Schema *>(this)->find(name);
 }
 
 Schema::iterator Schema::find(ObjectSchema const& object) noexcept
 {
-    auto it = std::lower_bound(begin(), end(), object, compare_by_name);
```

Comment:
Since `compare_by_name` is now only used in one place it might be worth getting rid of the function in favor of a lambda at the one place the comparison is needed.